server/enterprise/message_export/shared/shared.go

```diff
@@ -34,28 +61,28 @@ const (
 )
 
 // JobData keeps the current state of the job.
+// When used by a worker, all fields marked exported are exported to the job's job.Data prop bag)
 type JobData struct {
-	// If used by a worker, this section is saved in the job.Data field.
-	ExportType              string
-	ExportDir               string
-	BatchStartTime          int64
-	BatchStartId            string
+	ExportType              string // exported
+	ExportDir               string // exported
+	BatchStartTime          int64  // exported
+	BatchStartId            string // exported
 	ExportPeriodStartTime   int64
-	JobStartTime            int64
-	JobEndTime              int64
-	JobStartId              string
-	BatchSize               int
-	ChannelBatchSize        int
-	ChannelHistoryBatchSize int
-	BatchNumber             int
-	TotalPostsExpected      int
-	TotalPostsExported      int
+	JobStartTime            int64  // exported
+	JobEndTime              int64  // exported
+	JobStartId              string // exported
+	BatchSize               int    // exported
+	ChannelBatchSize        int    // exported
+	ChannelHistoryBatchSize int    // exported
+	BatchNumber             int    // exported
+	TotalPostsExpected      int    // exported
+	MessagesExported        int    // exported
```

Comment:
Hmm .. wondering if we can embed a struct inside JobData where all fields of that struct is written to the prop bag, and everything else declared in this struct is not. Or even better, we can probably declare separate structs. But not a blocker. It just seems additional overhead to keep maintaining a set of comments with the code.

Reply:
Good idea, I made a JobDataExported struct and I'm embedding it in the JobData struct. At least it keeps them separate and no need for comments (I didn't like that either, it was mostly for my own mental tracking).